cpu/esp_common/esp-wifi/esp_wifi_netdev.c

```diff
@@ -729,8 +794,9 @@ static wifi_config_t wifi_config_sta = {
 #endif
     }
 };
+#endif /* MODULE_ESP_WIFI_AP */
 
-#if defined(MCU_ESP8266) && !defined(MODULE_ESP_NOW)
+#if defined(MCU_ESP8266) && !defined(MODULE_ESP_NOW) && !defined(MODULE_ESP_WIFI_AP)
```

Comment:
Using 
```suggestion
#if (defined(MCU_ESP8266) && !defined(MODULE_ESP_NOW)) || defined(MODULE_ESP_WIFI_AP)
```
the same `wifi_config_ap` could be used for `esp_now` with esp8266 and `esp_wifi_ap`. You could than delete the line with `ssid_hidden = 1` and could define a conditional definition of these parameters, for example:
```c
#ifdef MODULE_ESP_WIFI_AP
        .ssid_hidden =ESP_WIFI_SSID_HIDDEN,     /* don't hide the AP s*/
        .max_connection = ESP_WIFI_MAX_CONN,    /* number of max connections  */
        .beacon_interval = 100,                 /* send beacon every 100 ms */
#else
        .ssid_hidden = 1,               /* don't make the AP visible */
        .max_connection = 0,            /* don't allow connections */
        .beacon_interval = 60000,       /* send beacon only every 60 s */
#endif
```
Then you could drop the second `wifi_config_ap`.

The comment of closing `#endif` would have to be updated:
```c
#endif /* defined(MCU_ESP8266) && !defined(MODULE_ESP_NOW)) || defined(MODULE_ESP_WIFI_AP) */
```